airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -449,6 +450,93 @@ def post_clear_task_instances(*, dag_id: str, session: Session = NEW_SESSION) ->
         TaskInstanceReferenceCollection(task_instances=task_instances.all())
     )
 
+@security.requires_access(
+    [
+        (permissions.ACTION_CAN_EDIT, permissions.RESOURCE_DAG),
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG_RUN),
+        (permissions.ACTION_CAN_DELETE, permissions.RESOURCE_TASK_INSTANCE),
+    ],
+)
+@provide_session
+def post_clear_task_dag_run_instances(
+    *, 
+    dag_id: str,
+    dag_run_id: str,
+    task_id: str, 
+    session: Session = NEW_SESSION,
+    ) -> APIResponse:
+    """Clear task instances for given dag run."""
+    dag = current_app.dag_bag.get_dag(dag_id)
```

Comment:
@bbovenzi Could you tell me if the checks are ok or do i have to make any changes?